src/Stratis.FederatedPeg.Tests/EventsPersisterTests.cs

```diff
@@ -1,87 +1,107 @@
 ﻿using System;
 using System.Linq;
 using System.Reactive.Linq;
-
-using NSubstitute;
 using System.Threading.Tasks;
-
-using DBreeze.Utils;
-
 using FluentAssertions;
-
 using Microsoft.Extensions.Logging;
-
-using NSubstitute.Core;
-
+using Moq;
+using NBitcoin;
+using NSubstitute;
+using Stratis.Bitcoin.Utilities;
 using Stratis.FederatedPeg.Features.FederationGateway.Interfaces;
 using Stratis.FederatedPeg.Features.FederationGateway.TargetChain;
 using Stratis.FederatedPeg.Tests.Utils;
 using Xunit;
 
-
 namespace Stratis.FederatedPeg.Tests
 {
     public class EventsPersisterTests : IDisposable
     {
+        private readonly Mock<ICrossChainTransferStore> mockStore;
```

Comment:
Curious as to why you changed that, there was examples of code doing the same call argument validation as you do...

Reply:
Also, I would have expected test for the new logic you added, rather than just changing from NSubstitute to Moq and making the old ones pass. Apologies if I missed them.

Reply:
The reason is that Mock seems to make it much easier to deal with setting up dynamic return values from properties. I'm not done adding test cases but your observations so far are welcomed.